other/pairs_sum_k/sean/find_pairs.py

```diff
@@ -0,0 +1,14 @@
+def find_pairs(nums, target):
+    # no duplicates
+    seen = set([])
+    pairs = set([])
+    for num in nums:
+        seen.add(num)
+    for num in nums:
+        compliment = target - num
+        pair = (num, compliment)
+        reversed_pair = (compliment, num)
+        if compliment in seen and reversed_pair not in pairs:
+            pairs.add(pair)
```

Comment:
This is interesting.  I think it's good the way it is, but it's worth talking about.

The way I've handled this in the past is to sort the pair before checking so that reversed pairs aren't ever added to the set.  I think both ways aren't particularly readable because there's this awkward step of saying "add this thing if neither the way it is nor the other way it could be were already added".

Okay I started looking things up in the middle of writing this and I've learned a thing.  It turns out you can't have sets of sets because sets aren't hashable, so you need [`frozenset`](https://docs.python.org/3/library/stdtypes.html#frozenset).  Basically you can have a set of sets as long as the inner sets are immutable, which lets us insert the pairs in either "order" because sets aren't ordered and get back to "add this thing if it wasn't added yet"